proxy/deploy.go

```diff
@@ -60,6 +61,7 @@ func DeployFunction(fprocess string, gateway string, functionName string, image
 	}
 
 	request, _ = http.NewRequest(method, gateway+"/system/functions", reader)
+	request.Header.Set("User-Agent", fmt.Sprintf("%s/%s", "FaaS-CLI", version.VersionStr()))
```

Comment:
We have some tests by @itscaro something like deploy_test.go or proxy_test.go - have a look at these, because we should extend to include this.

Reply:
I centralised in another branch that merges your proxy changes with requests. Once your changes are merged I'll amend this commit, but it'll make a mess to put into this.

Reply:
Also are the @itscaro tests the ones in [this commit](https://github.com/openfaas/faas-cli/commit/7e920f8a0857855781e740a8d3731a5a8a5d1747)? Otherwise I can't find them